File: Ejercicio 4/restaurante.py
from abc import ABC,abstractmethod

class Preparacion(ABC):
    @abstractmethod
    def preparar(self): pass

class Alimento:
    def __init__(self,nombre):
        self.nombre=nombre

class Hamburguesa(Alimento,Preparacion):
    def __init__(self): super().__init__("Hamburguesa")
    def preparar(self): print("Preparando hamburguesa en la Sarten")

class Pizza(Alimento,Preparacion):
    def __init__(self): super().__init__("Pizza")
    def preparar(self): print("Preparando pizza en el Horno")

class Ensalada(Alimento,Preparacion):
    def __init__(self): super().__init__("Ensalada")
    def preparar(self): print("Preparando ensalada en el Bowl")

menu=[Hamburguesa(),Pizza(),Ensalada()]

for a in menu:
    a.preparar()

File: Ejercicio 4/restaurante.java
interface Preparacion{
    void preparar();
}

abstract class Alimento{
    String nombre;

    Alimento(String n){ nombre=n; }
}

class Hamburguesa extends Alimento implements Preparacion{
    Hamburguesa(){ super("Hamburguesa"); }
    public void preparar(){ System.out.println("Preparando hamburguesa en la Sarten"); }
}

class Pizza extends Alimento implements Preparacion{
    Pizza(){ super("Pizza"); }
    public void preparar(){ System.out.println("Preparando pizza en el Horno"); }
}

class Ensalada extends Alimento implements Preparacion{
    Ensalada(){ super("Ensalada"); }
    public void preparar(){ System.out.println("Preparando ensalada en el Bowl"); }
}

public class Main{
    public static void main(String[] args){

        Preparacion[] menu={
            new Hamburguesa(),
            new Pizza(),
            new Ensalada()
        };

        for(Preparacion a:menu)
            a.preparar();
    }
}
